src/daemon/daemon.cpp

```diff
@@ -1083,81 +1083,126 @@ void mp::Daemon::find(const FindRequest* request, grpc::ServerWriterInterface<Fi
 try // clang-format on
 {
     mpl::ClientLogger<FindReply> logger{mpl::level_from(request->verbosity_level()), *config->logger, server};
+
     FindReply response;
+    response.set_show_images(request->show_images());
+    response.set_show_blueprints(request->show_blueprints());
+
     const auto default_remote{"release"};
 
     if (!request->search_string().empty())
     {
-        std::vector<std::pair<std::string, VMImageInfo>> vm_images_info;
-
-        try
-        {
-            vm_images_info = config->vault->all_info_for({"", request->search_string(), false, request->remote_name(),
-                                                          Query::Type::Alias, request->allow_unsupported()});
-        }
-        catch (const std::exception&)
+        if (request->show_images())
         {
             try
             {
-                vm_images_info.push_back(
-                    std::make_pair("", config->blueprint_provider->info_for(request->search_string())));
+                std::vector<std::pair<std::string, VMImageInfo>> vm_images_info =
+                    config->vault->all_info_for({"", request->search_string(), false, request->remote_name(),
+                                                 Query::Type::Alias, request->allow_unsupported()});
+
+                for (const auto& [remote, info] : vm_images_info)
+                {
+                    std::string name;
+                    if (info.aliases.contains(QString::fromStdString(request->search_string())))
+                    {
+                        name = request->search_string();
+                    }
+                    else
+                    {
+                        name = info.id.toStdString();
+                        name.resize(12);
+                    }
+
+                    auto entry = response.add_images_info();
+                    entry->set_os(info.os.toStdString());
+                    entry->set_release(info.release_title.toStdString());
+                    entry->set_version(info.version.toStdString());
+                    auto alias_entry = entry->add_aliases_info();
+                    if (!request->remote_name().empty() ||
+                        (request->remote_name().empty() && vm_images_info.size() > 1 && remote != default_remote))
+                    {
+                        alias_entry->set_remote_name(remote);
+                    }
+                    alias_entry->set_alias(name);
+                }
             }
             catch (const std::exception&)
             {
-                throw std::runtime_error(
```

Comment:
I didn't like the idea of throwing exceptions when nothing was found since not finding an image/blueprint doesn't necessarily mean that something went wrong. Most likely just that nothing was found. Perhaps the supporting `vault->all_info_for()` and `blueprint_provider->info_for()` functions should be changed to reflect this intuition since now we will silently eat any other errors that could happen.